### PR TITLE
New Feature: `Favourite addons`

### LINKED ISSUE #1250: Ability to make selected addons always show fist in the popup.
**Is your feature request related to a problem? Please describe.**

There are some addons (2.0 → 3.0, dark mode) that I want to be able to easily enable or disable, which can be harder if many addons are running on the current page so the list gets long.

**Describe the solution you'd like**

Ability to "pin" addons to the top of the addons tab.

### PR DESCRIPTION
**Pull request by Platformer444**
Resolves #1250 

### Changes
This enhancement allows certain addons to be favourited and be always show on the top.

### Reason for changes
Favouriting Addons allows easy enabling and disabling of the addon.

### Tests
Tested on Chrome 111 and it works fine.

### CODE REVIEW

File: images/icons/star-filled.svg
Comment (fractioneater) [inline images/icons/star-filled.svg:1]: That right "arm" is a bit off.
image

**Comment by WorldLanguages** [comment]
This is going in a good direction, but it will need many more changes and questions to reach users.
Firstly, we would store the "favorited" addons in chrome.storage, not localStorage. Should it be chrome.storage.sync or chrome.storage.local? I'm not sure. Would import/export also store the list of favorited addons?

**Comment by WorldLanguages** [comment]
Also, we use American English when naming variables, so multiple instances of "favourite" will need to get the U removed.

**Comment by Platformer444** [comment]
> This is going in a good direction, but it will need many more changes and questions to reach users. Firstly, we would store the "favorited" addons in chrome.storage, not localStorage. Should it be chrome.storage.sync or chrome.storage.local? I'm not sure. Would import/export also store the list of favorited addons?

I think `chrome.storage.sync` would be a better option as addon settings are also saved in it.

**Comment by Platformer444** [comment]
> > This is going in a good direction, but it will need many more changes and questions to reach users. Firstly, we would store the "favorited" addons in chrome.storage, not localStorage. Should it be chrome.storage.sync or chrome.storage.local? I'm not sure. Would import/export also store the list of favorited addons?
> 
> I think `chrome.storage.sync` would be a better option as addon settings are also saved in it.

@WorldLanguages

**Comment by TheColaber** [comment]
this needs discussion

**Pull request closed without merging**